oauth.go

```diff
@@ -24,6 +25,25 @@ type Host struct {
 	TokenURL      string
 }
 
+func NewGitHubHost(hostURL string) (*Host, error) {
+	base, err := url.Parse(strings.TrimSpace(hostURL))
+	if err != nil {
+		return nil, err
+	}
+
+	createURL := func(path string) string {
+		u := *base // Copy base URL
+		u.Path = path
+		return u.String()
+	}
+
+	return &Host{
+		DeviceCodeURL: createURL("/login/device/code"),
+		AuthorizeURL:  createURL("/login/oauth/authorize"),
+		TokenURL:      createURL("/login/oauth/access_token"),
+	}, nil
+}
+
 // GitHubHost constructs a Host from the given URL to a GitHub instance.
```

Comment:
Let's add a deprecation notice on the old function:

```suggestion
// GitHubHost constructs a Host from the given URL to a GitHub instance.
// 
// Deprecated: `GitHubHost` can panic with a malformed `hostURL`. Use `NewGitHubHost` instead for graceful error handling.
```